src/pg/result_set.cr

```diff
@@ -211,4 +211,12 @@ class PG::ResultSet < ::DB::ResultSet
     # if the connection is lost there is nothing to be
     # done since the result set is no longer needed
   end
+
+  private class Buffer < IO::Sized
+    getter connection : PG::Connection
+
+    def initialize(io, read_size, @connection)
+      super io, read_size
+    end
+  end
```

Comment:
I don't know if this needed to be `private`, but this was the only backwards-compatible way I could come up with to pass the connection to the decoder.